src/components/JobCard/__test__/JobCard.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render } from '@/tests/customRender'
+import { screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import JobCard from '../JobCard'
+
+// Mock data for the job
+const mockJob = {
+    title: 'Software Engineer',
+    company: 'Tech Corp',
+    location: 'Tokyo, Shibuya',
+    salary: '600,000 - 12,000,000',
+    tags: ['React', 'JavaScript', 'Frontend'],
+    logoUrl: 'https://via.placeholder.com/150',
+    publicationDate: '31/01/2025',
+    jobPostingUrl: 'https://jobpostingurl.com',
+}
+
+describe('JobCard Component', () => {
+    it('should render the job details correctly', () => {
+        render(<JobCard job={mockJob} />)
+
+        // Check the job title
+        expect(screen.getByText(mockJob.title)).toBeInTheDocument()
+
+        // Check the company name
+        expect(screen.getByText(mockJob.company)).toBeInTheDocument()
+
+        // Check the location
+        expect(screen.getByText(mockJob.location)).toBeInTheDocument()
+
+        // Check the salary
+        expect(screen.getByText(mockJob.salary)).toBeInTheDocument()
+
+        // Check the tags
+        mockJob.tags.forEach((tag) => {
+            expect(screen.getByText(tag)).toBeInTheDocument()
+        })
+
+        // Check the publication date
+        expect(screen.getAllByText(mockJob.publicationDate).length).toBeGreaterThan(0)
+
+        // Check the logo
+        const logo = screen.getByAltText(`${mockJob.company} logo`)
+        expect(logo).toBeInTheDocument()
+        expect(logo).toHaveAttribute('src', mockJob.logoUrl)
+    })
+
+    it('does not render salary if it is empty', () => {
+        const jobWithoutSalary = { ...mockJob, salary: '' }
+        render(<JobCard job={jobWithoutSalary} />)
+
+        // Ensure the salary is not rendered
+        expect(screen.queryByText(mockJob.salary)).not.toBeInTheDocument()
+    })
+
+    it('should renders tags with correct styles', () => {
+        const { container } = render(<JobCard job={mockJob} />)
+
+        // Select all Chip elements (MUI typically applies .MuiChip-root class)
+        const chips = container.querySelectorAll('.MuiChip-root')
+
```

Comment:
Selectors based on CSS tend to be very brittle and break as the codebase evolves. Also, we want to follow the [Testing Library Guiding Principles](https://testing-library.com/docs/guiding-principles): "The more your tests resemble the way your software is used, the more confidence they can give you."

So I usually try to select by
1. Role, which gives a semantic meaning
2. Text
3. If neither of the above exist, I may add an `aria-label="chip"` inside the `<Chip>` tag and then call 

```suggestion
        render(<JobCard job={mockJob} />)

        const chips = await screen.findAllByLabelText('chip')
```